### PR TITLE
Add test consent requirements of sideloaded EPUB publications

### PR DESCRIPTION
**Pull request by GJFR**
Add test for checking whether sideloaded EPUB publications require user consent for scripting and loading of remote resources (#297, [spec](https://www.w3.org/TR/epub-rs-33/#security-privacy-recommendations)).

- I split this test into `_scripting` and `_network`, but of course, these can be merged if that would be more convenient.
- There is a plethora of ways to instantiate remote connections, but to keep it simple I only included the resource types that are explicitly mentioned in the spec.
  - I used an audio file hosted by MDN, because I couldn't find one hosted by W3C. 
  - In my opinion, we should include a remote script as well, however, I could not find a suitable one hosted by W3C. Is there a possibility to host a small script on the W3C domain?

### CODE REVIEW

File: tests/sec-untrusted-consent_network/EPUB/content_001.xhtml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <p>Refers to <code>"https://www.w3.org/TR/epub-rs-33/"</code></p>
+    <iframe width="100" style="border: solid;" src="https://www.w3.org/TR/epub-rs-33/"></iframe>
+
+    <h4>Image</h4>
+    <p>Refers to <code>"https://www.w3.org/StyleSheets/TR/2021/logos/W3C"</code></p>
+    <img width="150" src="https://www.w3.org/StyleSheets/TR/2021/logos/W3C"/>
+
+    <h4>Audio</h4>
+    <p>Refers to <code>"https://interactive-examples.mdn.mozilla.net/media/cc0-audio/t-rex-roar.mp3"</code></p>
+    <audio controls="" src="https://interactive-examples.mdn.mozilla.net/media/cc0-audio/t-rex-roar.mp3">
```
Comment (iherman) [inline tests/sec-untrusted-consent_network/EPUB/content_001.xhtml:19]:
```suggestion
    <audio controls="" src="https://www.w3.org/2023/03/epub_tests_files/t-rex-roar.mp3">
```

File: tests/sec-untrusted-consent_network/EPUB/content_001.xhtml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    </audio>
+
+    <h4>Video</h4>
+    <p>Refers to <code>"https://interactive-examples.mdn.mozilla.net/media/cc0-videos/flower.webm"</code></p>
+    <video width="150" src="https://interactive-examples.mdn.mozilla.net/media/cc0-videos/flower.webm"></video>
```
Comment (iherman) [inline tests/sec-untrusted-consent_network/EPUB/content_001.xhtml:24]:
```suggestion
    <video width="150" src="https://www.w3.org/2023/03/epub_tests_files/flower.webm"></video>
```

**Review by iherman — COMMENTED** [review]
I have copied those files and put them on the W3C site, see my comments.

If you send me a script you have in mind, I am happy to put them on the W3C web site as well.

**Comment by GJFR** [comment]
Perfect, thanks!
This script should be sufficient:
```js
function test_remote_scripting_support() {
  const report = document.getElementById('remote_scripting_support');
  report.textContent = 'This reading system fetched and executed a remote script.';
}
window.addEventListener('load', test_remote_scripting_support);
```

File: tests/sec-untrusted-consent_network/EPUB/content_001.xhtml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>Fetching of remote resources embedded by side-loaded EPUB publications requires consent</title>
+    <link rel="stylesheet" href="https://www.w3.org/StyleSheets/TR/2021/W3C-CR"/>
+    <script src="!TODO"></script>
```
Comment (iherman) [inline tests/sec-untrusted-consent_network/EPUB/content_001.xhtml:5]:
```suggestion
    <script src="https://www.w3.org/2023/03/epub_tests_files/remote_scripting.js"></script>
```

**Review by iherman — APPROVED** [review]
I have added your script to the W3C Website, see change request. 

Like for the others, I am happy to merge once the implementation reports are also in place.

**Comment by GJFR** [comment]
Currently we check whether stylesheets are fetched and loaded without consent by simply referring to the spec's stylesheet, but this is not ideal (and this stylesheet may be subject to change in the future).
Could you please host the stylesheet below as well? I think this would make the tests more future-proof and robust.
```css
.paint_red {
  color: red
}
```
It's the last one, I promise 😉

File: tests/sec-untrusted-consent_network/EPUB/content_001.xhtml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <audio controls="" src="https://www.w3.org/2023/03/epub_tests_files/t-rex-roar.mp3">
+    </audio>
+
+    <h4>Video</h4>
+    <p>Refers to <code>"https://interactive-examples.mdn.mozilla.net/media/cc0-videos/flower.webm"</code></p>
+    <video width="150" src="https://www.w3.org/2023/03/epub_tests_files/flower.webm"></video>
+
+    <h4>Stylesheet</h4>
+    <p>Refers to <code>"https://www.w3.org/StyleSheets/TR/2021/W3C-CR"</code>.</p>
+    <p class="paint_red">Stylesheet is fetched if this text is red.</p>
```
Comment (iherman) [inline tests/sec-untrusted-consent_network/EPUB/content_001.xhtml:29]:
```suggestion
    <p>Refers to <code>"https://www.w3.org/2023/03/epub_tests_files/paint_red.css"</code>.</p>
    <p class="paint_red">Stylesheet is fetched if this text is red.</p>
```

File: tests/sec-untrusted-consent_network/EPUB/content_001.xhtml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xml:lang="en">
+  <head>
+    <title>Fetching of remote resources embedded by side-loaded EPUB publications requires consent</title>
+    <link rel="stylesheet" href="!TODO"/>
```
Comment (iherman) [inline tests/sec-untrusted-consent_network/EPUB/content_001.xhtml:4]:
```suggestion
    <link rel="stylesheet" href="https://www.w3.org/2023/03/epub_tests_files/paint_red.css"/>
```

**Review by iherman — APPROVED** [review]
Oops, I missed your last request... My apologies. Done now.

(Do not forget to regenerate the epub files themselves before declare this as done!)

**Pull request merged**